### PR TITLE
Adding initial metrics in the operator service

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

Adding initial set of metrics in operator service.

operator/service_created_total:  The total number of dapr service created
operator/service_deleted_total: The total number of dapr service deleted
operator/service_updated_total: The total number of dapr service updated

## Issue reference

#971

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: pkg/operator/monitoring/metrics.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		"The total number of dapr service deleted.",
+		stats.UnitDimensionless)
+	serviceUpdatedTotal = stats.Int64(
+		"operator/service_updated_total",
+		"The total number of dapr service updated.",
+		stats.UnitDimensionless)
+
+	noKeys = []tag.Key{}
+)
+
```
Comment (youngbupark) [inline pkg/operator/monitoring/metrics.go]: Can we add `app_id` tag to each metric? (tag term in oc is same as metric dimension)

Comment (shalabhms) [reply pkg/operator/monitoring/metrics.go]: done . please check.

Comment (shalabhms) [reply pkg/operator/monitoring/metrics.go]: done. please check.

**Comment by shalabhms** [comment]
image

cc @youngbupark , @yaron2 , @amanbha

File: pkg/operator/monitoring/metrics.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+const (
+	appID = "app_id"
+)
+
+var (
+	serviceCreatedTotal = stats.Int64(
+		"operator/service_created_total",
+		"The total number of dapr service created.",
```
Comment (yaron2) [inline pkg/operator/monitoring/metrics.go]: `dapr services`

File: pkg/operator/monitoring/metrics.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		stats.UnitDimensionless)
+	serviceDeletedTotal = stats.Int64(
+		"operator/service_deleted_total",
+		"The total number of dapr service deleted.",
```
Comment (yaron2) [inline pkg/operator/monitoring/metrics.go]: same as above, plural

File: pkg/operator/monitoring/metrics.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		stats.UnitDimensionless)
+	serviceUpdatedTotal = stats.Int64(
+		"operator/service_updated_total",
+		"The total number of dapr service updated.",
```
Comment (yaron2) [inline pkg/operator/monitoring/metrics.go]: plural services

File: pkg/operator/monitoring/metrics.go (hunk 2 above)
Comment (shalabhms) [reply pkg/operator/monitoring/metrics.go]: done.

File: pkg/operator/monitoring/metrics.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+func RecordServiceDeletedCount(appID string) {
+	stats.RecordWithTags(context.Background(), withTags(appIDKey, appID), serviceDeletedTotal.M(1))
+}
+
+// RecordServiceUpdatedCount records the number of dapr service updated
+func RecordServiceUpdatedCount(appID string) {
+	stats.RecordWithTags(context.Background(), withTags(appIDKey, appID), serviceUpdatedTotal.M(1))
+}
+
+// withTags converts tag key and value pairs to tag.Mutator array.
```
Comment (youngbupark) [inline pkg/operator/monitoring/metrics.go]: I think we need to move withTags util to common package :) we can do later.

Comment (shalabhms) [reply pkg/operator/monitoring/metrics.go]: sure.

**Pull request merged**